### PR TITLE
Added DONeRF project page and bibtex.

### PR DESCRIPTION
**Pull request by thomasneff**
# Pull requests instruction

1. Put the bibtex in `awesome-NeRF/citations/<YOUR-PROJECT>.txt`.
2. Modify the README and follow the format TITLE, AUTHOR, CONFERENCE YEAR | OPTIONAL LINK | BIBTEX. For example:
- [NeRF: Representing Scenes as Neural Radiance Fields for View Synthesis](https://www.matthewtancik.com/nerf), Mildenhall et al., ECCV 2020 | [github](https://github.com/bmild/nerf)  | [bibtex](./citations/nerf.txt)

### CODE REVIEW

**Comment by yenchenlin** [comment]
Thank you!!!

**Comment by thomasneff** [comment]
Thank you!
Sorry for not following the instructions, I was following the last PR's changelog since the link to the PR instructions in the README (https://github.com/yenchenlin/awesome-NeRF/wiki/Pull-Request-Steps) seems to be broken. Maybe you want to update that so other people won't get lost following the dead link 😄

**Comment by yenchenlin** [comment]
@thomasneff Oh yeah thanks for pointing that out, just fixed the README :)

**Comment by kwea123** [comment]
@thomasneff Thanks for your great work! Do you plan to release the code?

I have several questions:
1.  Sampling with known depth is efficient, but in real world scenes the depth is either sparse (e.g. LIDAR) or scale inconsistent (e.g. monodepth) and noisy. I think real depths probably go beyond the scope of the paper, but I think one interesting experiment is to see how the depth noise (GT) affects the image quality. I think only training on exact depth values (even if it's smoothed) is too restrictive.

2. NeRF also learns depth values without GT supervision, do you think it's possible to directly integrate your method into nerf's code? I.e. remove your oracle network, and place the samples according to the depth learnt by nerf?

If you plan to release the code, I can post the issue then again, thank you!

**Comment by thomasneff** [comment]
Hi @kwea123 !

(I think we should probably transition to https://github.com/DepthOracleNeRF/depthoraclenerf.github.io/issues for questions/issues regarding DONeRF so that we won't misuse this PR thread, but I'll answer your questions briefly here anyways. Feel free to create an issue in the other repo if you have any more questions!)

Before directly answering your questions, just to clarify: Our main goal with the DONeRF approach was to bring movie-like *generated* content to real-time using NeRF-like MLPs that can be efficiently compressed and potentially streamed, i.e., we wanted  a compact neural representation that is able to generate novel views for games or VR, where using the ground truth depth for training is typically no issue.  

0.) At this time, we don't have plans to release the code.

1.) That's a very good point, and is (at the current state of the paper) mostly untested. Some of our initial experiments suggest that, with some loss in quality and performance (requiring a bit more samples due potentially incorrect depth), it can still work, as long as the provided depth is mostly consistent between views and the images are calibrated/undistorted w.r.t. the camera model. Large scale inconsistencies between depth images of the same scene would be an issue, as you then wouldn't be able to train the depth oracle properly, so the version described in the paper at least expects that depth stays mostly consistent between different views. 

2.)  It is possible to extract either a depth-map or all of the opacities from NeRF and use that as a guide for sample placement or to train a depth oracle. However, there are two issues with that: First, the depth from NeRF can be arbitrarily incorrect as the network "cheats" to correctly show the view dependent effects in the RGB output, often showcasing artifacts of the RGB color in the depth output, which can be inconsistent. Secondly, placing samples "on-the-fly" based on the depth learned by NeRF wouldn't result in a speedup, as you'd need to query the MLP for all potential samples to figure out where the estimated depth is in the first place. You can "render" the estimated depth map or opacity values of a fully trained NeRF after training and use those for local sampling without an oracle, but then you'd have issues with generating novel views that are not in the training set (requiring some interpolation / projection of depth samples into the scene to get depth values), in addition to requiring a ton of memory/storage compared to the compact depth oracle MLP. There are some ways to potentially get around those limitations, which we'll be investigating going forward. 

I hope this addresses some of your questions, I'm happy to answer more over in the other repo if you have any 😄
